### ISSUE TITLE
Won't start after updating to macOS Sierra

### ISSUE THREAD

**Issue by dippnerd** [Siddharth11/iTunes-Remote#4]
I have a Mac Mini that I hadn't gotten around to updating to macOS Sierra from El Capitan when I discovered this project, so I installed it and it ran (mostly, the playlist controls were blank). Later I decided it was a good time to upgrade to Sierra, let it do its thing, but when I came back to the Mac and restarted this project, it errored out with the following:

> itunes_remote@1.0.0 start /Users/aaron/iTunes-Remote
> node server/server

/Users/aaron/iTunes-Remote/server
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::5000
    at Object.exports._errnoException (util.js:1036:11)
    at exports._exceptionWithHostPort (util.js:1059:20)
    at Server._listen2 (net.js:1252:14)
    at listen (net.js:1288:10)
    at Server.listen (net.js:1384:5)
    at Object.<anonymous> (/Users/aaron/iTunes-Remote/server/server.js:18:6)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)

npm ERR! Darwin 16.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v6.6.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! itunes_remote@1.0.0 start: `node server/server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the itunes_remote@1.0.0 start script 'node server/server'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the itunes_remote package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server/server
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs itunes_remote
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls itunes_remote
npm ERR! There is likely additional logging output above.

[npm-debug.txt](https://github.com/Siddharth11/iTunes-Remote/files/514838/npm-debug.txt)

**Comment by Siddharth11** [comment]
I'm also using macOS Sierra and it's working perfectly for me. This seems to be the issue on your device.

> Error: listen EADDRINUSE :::5000

Port 5000 is already in use. Try changing the port no. from **5000** to any other value say **3000** on the **server/server.js** file. Let me know if the issue still persist.

**Comment by dippnerd** [comment]
Wow I feel like an idiot! I didn't even see that port complaint. Sure enough, I changed it and it's working. Not sure what app changed since the update, but at least it was an easy fix, thanks! 

A bit unrelated, but I still don't get anything popping up from the buttons in the top corners, a panel slides over but its just blank, any ideas?

**Comment by Siddharth11** [comment]
It is supposed to show you all custom made playlists on the left and all the tracks in current playlist on the right. You can check the console too for its output. I'm not able to recreate it.

**Comment by dippnerd** [comment]
Is there anything I have to enable or tag playlists with? My console just shows "Connected to a client" and "Disconnected" over and over, that's it. I'm on the latest version of iTunes if it matters

**Comment by Siddharth11** [comment]
I'm not sure. For me it didn't require any configuration. Try to restart the server and console the output from **server/serverSideEvents.js** from functions emitting **getPlaylists** and **getTracksData**

**Comment by dippnerd** [comment]
I think I figured it out. I ran the individual "tell application" commands in the script editor and noticed it doesn't consider the "up next" queue as a playlist, so it returns my entire library (which I assume is just either too much data or I just wasn't waiting long enough for it to return). I ran a playlist directly in iTunes and those songs showed up on the right as expected. I had a feeling, Up Next is always causing problems for some reason. 

As for the playlists, it turns out I had a playlist with the ’ character in the name (not to be confused with ' unfortunately) and it didn't like that. Once I spotted it in the JSON I replaced it and voila!

This is an awesome tool, kudos on making it! iTunes control has been the missing link for a home automation control panel I'm trying to write for my iPad, I'll be sure to post the code once I have it in a decent state, if you have any interest :)
